alerts/alerts.go

```diff
@@ -22,413 +22,119 @@ import (
-		"downloadedBytes": payload.DownloadedBytes,
-		"totalBytes":      payload.TotalBytes,
-	}).Debug(payload.Message)
-
-	instance.addAlert(cloudprotocol.AlertItem{
-		Timestamp: time,
-		Tag:       cloudprotocol.AlertTagAosCore,
-		Source:    source,
-		Payload:   payload,
-	})
-}
-
-func (instance *Alerts) setupJournal(ctx context.Context) (err error) {
-	if instance.cursorStorage == nil {
-		return aoserrors.New("cursor storage is not set")
-	}
-
-	journal, err := instance.prepareJournal()
-	if err != nil {
-		return aoserrors.Wrap(err)
+		return
 	}
 
-	go func() {
-		storeTicker := time.NewTicker(storeCursorPeriod)
-		journalChanged := false
-		result := sdjournal.SD_JOURNAL_APPEND
-
-		for {
-			select {
-			case <-storeTicker.C:
-				if journalChanged {
-					journalChanged = false
-
-					if err = instance.storeCurrentCursor(journal); err != nil {
-						log.Errorf("Can't store journal cursor: %s", err)
-					}
-				}
-
-			case <-ctx.Done():
-				journal.Close()
-				return
-
-			default:
-				if result != sdjournal.SD_JOURNAL_NOP {
-					journalChanged = true
-
-					if err = instance.processJournal(journal); err != nil {
-						log.Errorf("Journal process error: %s", err)
-					}
-				}
-
-				if result = journal.Wait(waitJournalTimeout); result < 0 {
-					log.Errorf("Wait journal error: %s", syscall.Errno(-result))
-				}
-			}
-		}
-	}()
-
-	return nil
+	instance.alertsChannel <- alert
 }
 
-func (instance *Alerts) prepareJournal() (journal *sdjournal.Journal, err error) {
-	journal, err = sdjournal.NewJournal()
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddMatch("PRIORITY=0"); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddMatch("PRIORITY=1"); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddMatch("PRIORITY=2"); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddMatch("PRIORITY=3"); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddDisjunction(); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.AddMatch("_SYSTEMD_UNIT=init.scope"); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if err = journal.SeekTail(); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if _, err = journal.Previous(); err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	cursor, err := instance.cursorStorage.GetJournalCursor()
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
-	}
-
-	if cursor != "" {
-		if err = journal.SeekCursor(cursor); err != nil {
-			return nil, aoserrors.Wrap(err)
-		}
+func (instance *Alerts) processAlertChannels() {
+	sendTicker := time.NewTicker(instance.config.SendPeriod.Duration)
 
-		if _, err = journal.Next(); err != nil {
-			return nil, aoserrors.Wrap(err)
-		}
-	} else {
-		if err = instance.storeCurrentCursor(journal); err != nil {
-			return nil, aoserrors.Wrap(err)
-		}
-	}
+	ctx, cancelFunction := context.WithCancel(context.Background())
```

Comment:
Move creating context and assigning cenacelFunction to New and pass context as parameter here. Otherwise, there is potential race.